dmoj/checkers/floats.py

```diff
@@ -1,24 +1,24 @@
 def check(process_output, judge_output, precision, **kwargs):
-    from itertools import izip
-    
+    from six.moves import zip
+
     # Discount empty lines
-    process_lines = filter(None, process_output.split('\n'))
-    judge_lines = filter(None, judge_output.split('\n'))
-    
+    process_lines = list(filter(None, process_output.split(b'\n')))
+    judge_lines = list(filter(None, judge_output.split(b'\n')))
```

Comment:
`six.moves.filter` is `itertools.ifilter`, we need `len(*_lines)` later on so it needs to be consumed.

Reply:
The problem is that you are creating a new list on python 2. I don't know if it's smart enough to optimize.

Reply:
Sure, that is a problem, but `six.moves.filter` is not the solution. Then again it might not actually matter, since big files we leave to the C checker anyway, and a couple thousand lines here would not be horrifyingly bad.

Reply:
??? `list(six.moves.filter(...))`